src/main/java/uk/gov/hmcts/reform/feature/security/CustomUserPermissionsFilter.java

```diff
@@ -4,54 +4,64 @@
 import org.springframework.security.core.userdetails.User;
 import org.springframework.security.core.userdetails.UserDetails;
-import org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter;
 import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+import org.springframework.web.filter.GenericFilterBean;
 import uk.gov.hmcts.reform.feature.model.UserRoles;
 
 import java.io.IOException;
 import java.util.List;
+import javax.servlet.FilterChain;
 import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
 import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
 
 import static org.springframework.http.HttpMethod.GET;
 
-public class CustomUserRolesFilter extends AbstractAuthenticationProcessingFilter {
+public class CustomUserPermissionsFilter extends GenericFilterBean {
 
-    static final String USER_ID_HEADER = "X-USER-ID";
+    public static final String USER_ID_HEADER = "X-USER-ID";
 
-    static final String USER_ROLES_HEADER = "X-USER-ROLES";
+    public static final String USER_PERMISSIONS_HEADER = "X-USER-PERMISSIONS";
 
-    public CustomUserRolesFilter(String pattern) {
-        super(new AntPathRequestMatcher(pattern, GET.name()));
+    private final RequestMatcher matcher;
+
+    public CustomUserPermissionsFilter(String pattern) {
+        matcher = new AntPathRequestMatcher(pattern, GET.name());
     }
 
     @Override
-    public Authentication attemptAuthentication(
-        HttpServletRequest request,
-        HttpServletResponse response
-    ) throws AuthenticationException, IOException, ServletException {
-        String userIdHeader = request.getHeader(USER_ID_HEADER);
-        String userRolesHeader = request.getHeader(USER_ROLES_HEADER);
-        Authentication originalAuth = SecurityContextHolder.getContext().getAuthentication();
-
-        if (checkHeadersAreValid(userIdHeader, userRolesHeader)) {
-            UserRoles userRoles = new UserRoles(
-                userIdHeader,
-                userRolesHeader.split(",")
-            );
-
-            return parseUserRoles(userRoles, originalAuth);
+    public void doFilter(
+        ServletRequest request,
+        ServletResponse response,
+        FilterChain chain
+    ) throws IOException, ServletException {
+        HttpServletRequest httpRequest = (HttpServletRequest) request;
+
+        if (matcher.matches(httpRequest)) {
+            String userIdHeader = httpRequest.getHeader(USER_ID_HEADER);
+            String userRolesHeader = httpRequest.getHeader(USER_PERMISSIONS_HEADER);
+            Authentication originalAuth = SecurityContextHolder.getContext().getAuthentication();
+
+            if (checkHeadersAreValid(userIdHeader, userRolesHeader)) {
+                UserRoles userRoles = new UserRoles(
```

Comment:
Class name is `UserRoles`  but looks like this is a User/UserDetails object with id and list of roles ?

Reply:
Wanted to but `UserDetails` already in scope. Since it only contained user roles, named it ...`UserRoles`